modules/zuora/src/billingPreview.ts

```diff
@@ -21,22 +21,55 @@ export const getBillingPreview = async (
 	return zuoraClient.post(path, body, billingPreviewSchema);
 };
 
-export const getNextInvoiceItems = (
-	billingPreview: BillingPreview,
-): InvoiceItem[] => {
-	const sorted = billingPreview.invoiceItems.sort((a, b) => {
-		return a.serviceStartDate < b.serviceStartDate ? -1 : 1;
-	});
-	const nextInvoiceDate = getIfDefined(
-		sorted[0]?.serviceStartDate.toISOString(),
-		'No invoice items found in response from Zuora',
+export function getNextInvoiceTotal(
+	invoiceItems: Array<{ date: Date; amount: number }>,
+) {
+	const ordered = getOrderedInvoiceData(invoiceItems);
+
+	return getIfDefined(
+		ordered[0],
+		'could not find a payment in the invoice preview',
 	);
+}
+
+export function getNextNonFreePaymentDate(
+	invoiceItems: Array<{ date: Date; amount: number }>,
+) {
+	const ordered = getOrderedInvoiceData(invoiceItems);
 
-	const groupedInvoiceItems = groupBy(billingPreview.invoiceItems, (item) =>
-		item.serviceStartDate.toISOString(),
+	const firstNonFree = getIfDefined(
+		ordered.find((item) => item.total > 0),
+		'could not find a non free payment in the invoice preview',
 	);
-	return getIfDefined(
-		groupedInvoiceItems[nextInvoiceDate],
-		'No invoice items found for next invoice date',
+
+	const nextPaymentDate = zuoraDateFormat(dayjs(firstNonFree.date));
+
+	return nextPaymentDate;
+}
+
+function getOrderedInvoiceData(
+	invoiceItems: Array<{ date: Date; amount: number }>,
+) {
+	const grouped = groupBy(invoiceItems, (invoiceItem) => {
+		return zuoraDateFormat(dayjs(invoiceItem.date));
+	});
+
+	const ordered = sortBy(Object.entries(grouped), (item) => item[0]).map(
+		(item) => {
+			return {
+				date: new Date(item[0]),
+				total: sumNumbers(item[1].map((item) => item.amount)),
+			};
+		},
 	);
-};
+	return ordered;
+}
+
+export function billingPreviewToRecords(billingPreviewAfter: BillingPreview) {
+	return billingPreviewAfter.invoiceItems.map((entry) => {
+		return {
+			date: entry.serviceStartDate,
+			amount: entry.chargeAmount + entry.taxAmount,
+		};
+	});
```

Comment:
If you wanted to, you could omit the `return` keyword in a simple arrow function like this:

```suggestion
	return billingPreviewAfter.invoiceItems.map((entry) => ({
		date: entry.serviceStartDate,
		amount: entry.chargeAmount + entry.taxAmount,
	}));
```